src/doc/rgw/multisite-reshard.md

```diff
@@ -66,15 +69,22 @@ The distinction between *index layout* and *log layout* is important, because in
 * For backward compatibility, add special handling when we get ENOENT trying to read this per-bucket sync status:
     - If the remote's oldest log layout has generation=0, read any existing per-shard sync status objects. If any are found, resume incremental sync from there.
     - Otherwise, initialize for full sync.
+* Use cls_version on the sync status object to detect racing writes. On ECANCELED:
+    - Re-read the sync status object.
+    - If the stored generation number no longer matches (i.e. raced with 'bucket sync init'), return the error.
+    - Otherwise retry the write until it succeeds.
+* Add a generation number to the per-shard sync status objects so we can reuse the same shard objects between generations. Bucket sync just ignores markers from older generations.
 
 ### Bucket Sync
 
 * Full sync uses a single bucket-wide listing to fetch all objects.
     - Use a cls_lock to prevent different shards from duplicating this work.
 * When incremental sync gets to the end of a log shard (i.e. listing the log returns truncated=false):
     - If the remote has a newer log generation, flag that shard as 'done' in the bucket sync status.
-    - Once all shards in the current generation reach that 'done' state, incremental bucket sync can advance to the next generation.
-    - Use cls_version on the bucket sync status object to detect racing writes from other shards.
+    - Once all shards in the current generation reach that 'done' state, incremental bucket sync can transition to the next generation.
+* On transition:
+    - Clear the set of 'done' shards and update the num_shards and generation in the bucket sync status.
+    - If num_shards was reduced, delete any extra per-shard sync status objects.
```

Comment:
@cbodley sorry I had missed looking at this pr earlier. what does extra status object mean? once a shard has completed syncing for a generation, we just have to delete that shard status object right?